platform/swift/source/src/bridge.rs

```diff
@@ -567,6 +581,126 @@ extern "C" fn capture_process_issue_reports(
   );
 }
 
+#[no_mangle]
+extern "C" fn capture_persist_javascript_error_report(
+  logger_id: LoggerId<'_>,
+  error_name: *const c_char,
+  error_message: *const c_char,
+  stack_trace: *const c_char,
+  is_fatal: bool,
+  engine: *const c_char,
+  debugger_id: *const c_char,
+  timestamp_millis: i64,
+  destination_path: *const c_char,
+) {
+  with_handle_unexpected(
+    || -> anyhow::Result<()> {
+      let error_name_str = unsafe { CStr::from_ptr(error_name) }.to_str()?;
```

Comment:
in Rust you typically wouldn't put the typename at the end like this, shadowing is much more idiomatic

```
let error_name = unsafe { CStr::from_ptr(err_name)}.to_str()?;
```